### PR TITLE
Fixes inconsistent initialization in compileBrTable test

### LINKED ISSUE #328: Flake on jit_test.go: TestAmd64Compiler_compileBrTable
at the time, the code that failed was this line, which was nil.

```
	jitcall(
		uintptr(unsafe.Pointer(&code[0])),
		uintptr(unsafe.Pointer(j.vm)),
	)
```

It should not be nil due to jit_amd64.s
```
// jitcall(codeSegment, callEngine)
TEXT ·jitcall(SB),NOSPLIT|NOFRAME,$0-16
--snip--
```

```
--- FAIL: TestAmd64Compiler_compileBrTable (0.39s)
[18](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:18)
    --- FAIL: TestAmd64Compiler_compileBrTable/index_register=2074,tmpRegister=2064 (0.02s)
[19](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:19)
        --- FAIL: TestAmd64Compiler_compileBrTable/index_register=[20](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:20)74,tmpRegister=2064/select_last_with_five_targets (0.01s)
20
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
[21](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:21)
	panic: runtime error: invalid memory address or nil pointer dereference
[22](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:22)
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x24ae066]
[23](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:23)

[24](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:24)
goroutine 862 [running]:
[25](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:25)
testing.tRunner.func1.2({0x12ae360, 0x1513370})
[26](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:26)
	/Users/runner/hostedtoolcache/go/1.17.7/x64/src/testing/testing.go:1209 +0x24e
[27](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:27)
testing.tRunner.func1()
[28](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:28)
	/Users/runner/hostedtoolcache/go/1.17.7/x64/src/testing/testing.go:1212 +0x218
[29](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:29)
panic({0x12ae360, 0x1513370})
[30](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:30)
	/Users/runner/hostedtoolcache/go/1.17.7/x64/src/runtime/panic.go:1038 +0x215
[31](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:31)
github.com/tetratelabs/wazero/internal/wasm/jit.(*jitEnv).exec(0xc00000e078, {0x24ae000, 0x1a9, 0x1a9})
[32](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:32)
	/Users/runner/work/wazero/wazero/internal/wasm/jit/jit_test.go:122 +0x20e
[33](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:33)
github.com/tetratelabs/wazero/internal/wasm/jit.TestAmd64Compiler_compileBrTable.func1(0x1361978, 0xc00009ad80, 0x12abe40)
```

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Also fixes some residual use of the word "vm"

### CODE REVIEW

File: internal/wasm/jit/jit_arm64.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func newArchContext() archContext {
 type archContext struct {
 	// jitCallReturnAddress holds the absolute return address for jitcall.
 	// The value is set whenever jitcall is executed and done in jit_arm64.s
-	// Native code can return back to the vm.exec's main loop back by
+	// Native code can return back to the ce.exec's main loop back by
```
Comment (codefromthecrypt) [inline internal/wasm/jit/jit_arm64.go:40]:
```suggestion
	// Native code can return back to the ce.execWasmFunction's main loop back by
```

File: internal/wasm/jit/jit_amd64_test.go [hunk 2]
```diff
@@ -345,7 +345,6 @@ func TestAmd64Compiler_compileBrTable(t *testing.T) {
 		require.NoError(t, err)
 
 		// Run codes
-		env := newJITEnvironment()
```
Comment (codefromthecrypt) [inline internal/wasm/jit/jit_amd64_test.go:348]: bug

**Review by codefromthecrypt — COMMENTED** [review]
notes!

File: internal/wasm/jit/jit_amd64_test.go (hunk 2 above)
Comment (mathetake) [reply internal/wasm/jit/jit_amd64_test.go:348]: I don't believe this is the cause of flakiness (as jitEnvironment has nothing to do with the code in this case) but at least this is inconsistent

**Review by mathetake — APPROVED** [review]

File: internal/wasm/jit/jit_amd64_test.go (hunk 2 above)
Comment (mathetake) [reply internal/wasm/jit/jit_amd64_test.go:348]: inserting `.Parallel()` under `t.Run(fmt.Sprintf("index_register=%d,tmpRegister=%d", indexReg, tmpReg)` causes similar segfalult so this must be something to do race

**Comment by codefromthecrypt** [comment]
okie dokie. I'll leave it at an inconsistency fix and untag the other issue. I was hoping it would have fixed the crash which has been more consistently failing recently.

File: internal/wasm/jit/jit_amd64_test.go (hunk 2 above)
Comment (codefromthecrypt) [reply internal/wasm/jit/jit_amd64_test.go:348]: I was wondering if there was a way to pre-initialize things. In java sometimes we eagerly resolve a reference to ensure no late pointers at runtime (ex static inner types we end up having to do this sometimes)

**Pull request merged**